### ISSUE TITLE
Calling revalidateTag() in an action causes the page to be rerendered even if it doesn't use the tag

### ISSUE THREAD

**Issue by fitimbytyqi** [vercel/next.js#71612]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/n9sttx

### To Reproduce

The sandbox above will not work because I've hidden the api for security purposes, but If you have an api that does those 2 requests you can replicate the issue.

1. Click on the block/unblock button

### Current vs. Expected behavior

Following the steps you'll see the revalidateTag will work even though there is no tag associated with that request.
Based on the documentation, revalidateTag revalidates only a specific tag that you put as an argument, but in this case it's refetching the rsc even though the tag doesn't exist.

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Sun Aug  6 20:05:33 UTC 2023
  Available memory (MB): 4102
  Available CPU cores: 2
Binaries:
  Node: 20.11.1
  npm: 10.2.4
  Yarn: 1.22.19
  pnpm: 8.15.4
Relevant Packages:
  next: 14.2.1 // An outdated version detected (latest is 15.0.0), upgrade is highly recommended!
  eslint-config-next: 14.2.1
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
 ⚠ An outdated version detected (latest is 15.0.0), upgrade is highly recommended!
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next dev (local), next start (local)

### Additional context

I tested this in a fresh next 15 setup and it acted the same.
Here's the discord issue link as well https://discord.com/channels/752553802359505017/1297903434808430612

**Comment by columk1** [comment]
Calling `revalidateTag` from an action will invalidate the router cache and re-render the current route, regardless of the tag. It's like calling `router.refresh()` from the client. See here: [cached data](https://nextjs.org/docs/app/building-your-application/caching). If you call it from a route handler this doesn't happen. The main docs page for revalidateTag doesn't mention this behaviour.

If you cache that fetch with "force-cache" and add a tag, you should still see the stale data after the refresh, or fresh data if the correct tag is used to revalidate.

**Comment by fitimbytyqi** [comment]
@columk1 Thanks for the explanation! That makes more sense now as to why adding any tag to the revalidationTag would invalidate the whole cache.
